File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/GenerationsStandingSignBlock.java
package generations.gg.generations.core.generationscore.world.level.block;

import generations.gg.generations.core.generationscore.world.level.block.entities.GenerationsSignBlockEntity;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.block.StandingSignBlock;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.block.state.properties.WoodType;
import org.jetbrains.annotations.NotNull;

public class GenerationsStandingSignBlock extends StandingSignBlock {

    public GenerationsStandingSignBlock(Properties arg, WoodType arg2) {
        super(arg, arg2);
    }

    @Override
    public BlockEntity newBlockEntity(@NotNull BlockPos arg, @NotNull BlockState arg2) {
        return new GenerationsSignBlockEntity(arg, arg2);
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/GenerationsStandingSignBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsSignBlockEntity
import net.minecraft.core.BlockPos
import net.minecraft.world.level.block.StandingSignBlock
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.block.state.properties.WoodType

class GenerationsStandingSignBlock(arg2: WoodType, arg: Properties) :
    StandingSignBlock(arg2, arg) {
    override fun newBlockEntity(arg: BlockPos, arg2: BlockState): BlockEntity? {
        return GenerationsSignBlockEntity(arg, arg2)
    }
}
